kong/db/schema/init.lua

```diff
@@ -1746,18 +1746,35 @@ local function validate_deprecation_exclusiveness(data, shorthand_value, shortha
   if shorthand_value == nil or
       shorthand_value == ngx.null or
       shorthand_definition.deprecation == nil or
-      shorthand_definition.deprecation.replaced_with == nil then
+      (shorthand_definition.deprecation.replaced_with == nil and shorthand_definition.translate_backwards == nil) then
     return true
   end
 
-  for _, replaced_with_element in ipairs(shorthand_definition.deprecation.replaced_with) do
-    local new_field_value = replaced_with_element.reverse_mapping_function and replaced_with_element.reverse_mapping_function(data)
-                                                                            or table_path(data, replaced_with_element.path)
+  if shorthand_definition.deprecation.replaced_with then
+    for _, replaced_with_element in ipairs(shorthand_definition.deprecation.replaced_with) do
+      local new_field_value = replaced_with_element.reverse_mapping_function and replaced_with_element.reverse_mapping_function(data)
+                                                                              or table_path(data, replaced_with_element.path)
+
+      if new_field_value and
+        new_field_value ~= ngx.null and
+        not deepcompare(new_field_value, shorthand_value) then
+        local new_field_name = join_string(".", replaced_with_element.path)
+
+        return nil, string.format(
+          "both deprecated and new field are used but their values mismatch: %s = %s vs %s = %s",
+          shorthand_name, tostring(shorthand_value),
+          new_field_name, tostring(new_field_value)
+        )
+      end
+    end
+  elseif shorthand_definition.translate_backwards then
+    local new_field_value = shorthand_definition.translate_backwards_with and shorthand_definition.translate_backwards_with(data)
+                                                                          or table_path(data, shorthand_definition.translate_backwards)
 
     if new_field_value and
-      new_field_value ~= ngx.null and
-      not deepcompare(new_field_value, shorthand_value) then
-      local new_field_name = join_string(".", replaced_with_element.path)
+        new_field_value ~= ngx.null and
```

Comment:
would it make sense to turn this check into an early return?

Reply:
I meant instead of doing a long condition for the `if` maybe do:

```
if not new_field_value or new_field_value == ngx.null then
  return true
end
```

just to make it a bit more readable. Not sure if it's a good idea here though, you're more familiar with the logic to make the best decision - feel free to disregard, it's an extra minor comment.